# patch to fix python-3.12 compatibility, unreleased patch from upstream https://github.com/s3tools/s3cmd/commit/9ecb2d853c54f87c6a3ec0423986dcc1a80e4678
From 9ecb2d853c54f87c6a3ec0423986dcc1a80e4678 Mon Sep 17 00:00:00 2001
From: Florent Viard <florent@sodria.com>
Date: Fri, 20 Oct 2023 01:28:37 +0200
Subject: [PATCH] Fixes #1343 - bytes to be used for repl string in
 stripNameSpace

Fixes the following issue with python 3.12:
TypeError: sequence item 1: expected str instance, bytes found
---
 S3/BaseUtils.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/S3/BaseUtils.py b/S3/BaseUtils.py
index b6b79086..69590649 100644
--- a/S3/BaseUtils.py
+++ b/S3/BaseUtils.py
@@ -273,7 +273,7 @@ def stripNameSpace(xml):
     xmlns_match = RE_XML_NAMESPACE.match(xml)
     if xmlns_match:
         xmlns = xmlns_match.group(3)
-        xml = RE_XML_NAMESPACE.sub("\\1\\2", xml, 1)
+        xml = RE_XML_NAMESPACE.sub(b"\\1\\2", xml, 1)
     else:
         xmlns = None
     return xml, xmlns
